cmd/tools/rest/rest.go

```diff
@@ -347,62 +347,69 @@ func GetPollerAndAddr(pName string) (*conf.Poller, string, error) {
 
 // FetchForCli used for CLI only
 func FetchForCli(client *Client, href string, records *[]any, downloadAll bool, curls *[]string) error {
-	getRest, err := client.GetRest(href)
-	if err != nil {
-		return fmt.Errorf("error making request %w", err)
-	}
-
-	pollerAuth, err := client.auth.GetPollerAuth()
-	if err != nil {
-		return err
-	}
-	*curls = append(*curls, fmt.Sprintf("curl --user %s --insecure '%s%s'", pollerAuth.Username, client.baseURL, href))
 
-	isNonIterRestCall := false
-	value := gjson.GetBytes(getRest, "records")
-	if value.String() == "" {
-		isNonIterRestCall = true
-	}
+	var prevLink string
+	nextLink := href
 
-	if isNonIterRestCall {
-		contentJSON := `{"records":[]}`
-		response, err := sjson.SetRawBytes([]byte(contentJSON), "records.-1", getRest)
+	for {
+		getRest, err := client.GetRest(nextLink)
 		if err != nil {
-			return fmt.Errorf("error setting record %w", err)
+			return fmt.Errorf("error making request %w", err)
 		}
-		var page Pagination
-		err = json.Unmarshal(response, &page)
+
+		pollerAuth, err := client.auth.GetPollerAuth()
 		if err != nil {
-			return fmt.Errorf("error unmarshalling json %w", err)
+			return err
 		}
-		*records = append(*records, page.Records...)
-	} else {
-		// extract returned records since paginated records need to be merged into a single list
+		*curls = append(*curls, fmt.Sprintf("curl --user %s --insecure '%s%s'", pollerAuth.Username, client.baseURL, href))
+
```

Comment:
it should be `nextLink` instead of `href`

Reply:
good catch